### ISSUE TITLE
hash mismatch for grpcio-status-1.43.0.tar.gz

### ISSUE THREAD

**Issue by veprbl** [NixOS/nixpkgs#151647]
```
# nix-store -r /nix/store/gfl9pp2y3pvgz0yhkhk194sd8d8ws4g2-grpcio-status-1.43.0.tar.gz.drv
this derivation will be built:
  /nix/store/gfl9pp2y3pvgz0yhkhk194sd8d8ws4g2-grpcio-status-1.43.0.tar.gz.drv
building '/nix/store/gfl9pp2y3pvgz0yhkhk194sd8d8ws4g2-grpcio-status-1.43.0.tar.gz.drv'...

trying https://files.pythonhosted.org/packages/source/g/grpcio-status/grpcio-status-1.43.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  8686  100  8686    0     0  88266      0 --:--:-- --:--:-- --:--:-- 88266
error: hash mismatch in fixed-output derivation '/nix/store/gfl9pp2y3pvgz0yhkhk194sd8d8ws4g2-grpcio-status-1.43.0.tar.gz.drv':
         specified: sha256-JVM8TWXX1ROmOPDEuIsrZnAOO/Q+aZWlWjvRXsC3eQI=
            got:    sha256-IXWQBvNqf/v/GH1BkfQRjActiqn6aCOhGq14QqPGzNA=
```

cc @fabaff

**Comment by veprbl** [comment]
https://pypi.org/project/grpcio-status/#copy-hash-modal-dd2e85cd-86b9-4306-96d9-b3c4757607c4
reports hash to also be `sha256-IXWQBvNqf/v/GH1BkfQRjActiqn6aCOhGq14QqPGzNA=`

**Comment by veprbl** [comment]
Regression from 1fcc7c0ef44f208c095318a7485710392afcf0f4, cc @marsam

**Comment by marsam** [comment]
It happens because grpcio-status uses the version of grpc to fetch the package from pypi,
I think we can solve it using something like:

```diff
--- i/pkgs/development/python-modules/grpcio-status/default.nix
+++ w/pkgs/development/python-modules/grpcio-status/default.nix
@@ -10,15 +10,12 @@
 
 buildPythonPackage rec {
   pname = "grpcio-status";
-  inherit (grpc) version;
+  inherit (grpc) src version;
   format = "setuptools";
 
   disabled = pythonOlder "3.6";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "JVM8TWXX1ROmOPDEuIsrZnAOO/Q+aZWlWjvRXsC3eQI=";
-  };
+  sourceRoot = "source/src/python/grpcio_status";
 
   propagatedBuildInputs = [
     googleapis-common-protos
```

**Comment by veprbl** [comment]
I like the suggestion above, however this is now closed in #151666
